doc/sphinx/practical-tools/coqide.rst

```diff
@@ -5,130 +5,183 @@
+
+While synchronous mode stops processing at the first error it encounters, in async
+mode, errors only stop processing the proof the error appears in.
+Therefore it can report errors in multiple proofs without manual intervention.
+In addition, async mode lets the user edit failed proofs without invalidating
+successful proofs that appear after it in the script.  The editable part of the proof
+is the part that appears between `Proof.` and `Qed.`.
+
+.. image:: ../_static/async-mode.png
+   :alt: Async mode
+
+In the screenshot, the proof of the failed theorem can be edited (between `Proof.`
+and `Qed.`) without invalidating the theorems that follow it.  The modified
+proof can then be reprocessed using the usual navigation operations.  The light blue
+highlight in the script indicates commands that haven't been processed.
+
+Async mode defers the final type checking step of proofs, leaving the `Qed.` marked in
+a slightly different shade of light blue to indicate this.  To complete the final
+checking, click on the "gears" button on the toolbar ("Fully check the document").
 
 Commands and templates
 ----------------------
 
 The Templates menu allows using shortcuts to insert
-commands. This is a nice way to proceed if you are not sure of the
+commands. This is a nice way to proceed if you're not sure of the
 syntax of the command you want.
 
 Moreover, from this menu you can automatically insert templates of complex
 commands like ``Fixpoint`` that you can conveniently fill afterwards.
 
 Queries
-------------
+-------
 
 .. image:: ../_static/coqide-queries.png
    :alt: CoqIDE queries
 
-We call *query* any command that does not change the current state,
-such as ``Check``, ``Search``, etc. To run such commands interactively, without
-writing them in scripts, CoqIDE offers a *query pane*. The query pane can be
-displayed on demand by using the ``View`` menu, or using the shortcut ``F1``.
-Queries can also be performed by selecting a particular phrase, then choosing an
-item from the ``Queries`` menu. The response then appears in the message window.
-The image above shows the result after selecting of the phrase
-``Nat.mul`` in the script window, and choosing ``Print`` from the ``Queries``
-menu.
+A *query* is any command that does not change the current state, such as
+:cmd:`Check`, :cmd:`Search`, etc.  The *query pane* lets you run such commands
+interactively without modifying your script. The query pane is accessible from
+the *View* menu, or using the shortcut ``F1``.
+You can also do queries by selecting some text, then choosing an
+item from the *Queries* menu. The response will appear in the message panel.
+The image above shows the result after selecting
+``Nat.mul`` in the bottom line of the script panel, then choosing ``Print``
+from the ``Queries`` menu.
+
+.. todo: should names of menus be *Menu* or `Menu` or ??  not consistent
 
 
 Compilation
-----------------
+-----------
 
 The `Compile` menu offers direct commands to:
```

Comment:
Shall we mention that this doesn't work at the moment, might get fixed in the future and that compiling from the terminal is a more sound alternative?

Reply:
In this case, yes.  It's on my list to fix. I expect it will be straightforward and that I'll get to it fairly soon.